src/test/java/seedu/task/logic/EditCommandTest.java

```diff
@@ -6,32 +6,24 @@
 import seedu.task.logic.commands.EditTaskCommand;
 import seedu.task.model.TaskBook;
 import seedu.task.model.item.Task;
+import seedu.taskcommons.core.Messages;
 
 public class EditCommandTest extends CommandTest {
 
-    /* Need help with this test to test for duplicates not allowed. By running the program, the exception can be caught
-     * */
-
-    @Ignore
     @Test
     public void execute_editFloatTask_duplicate() throws Exception {
         // setup expectations
         TestDataHelper helper = new TestDataHelper();
         Task toBeAdded = helper.computingFloatTask();
         TaskBook expectedAB = new TaskBook();
         expectedAB.addTask(toBeAdded);
-        toBeAdded = helper.computingEditedNameFloatTask();
-        expectedAB.addTask(toBeAdded);
-        
-        // setup model
-        model.addTask(helper.computingFloatTask());
-        model.addTask(helper.computingEditedNameFloatTask());
+        Task toBeAdded2 = helper.computingEditedNameFloatTask();
+        expectedAB.addTask(toBeAdded2);
         Task toBeEdited = helper.computingFloatTask();
-        model.editTask(toBeEdited, 0);
 
         // execute command and verify result
-        assertEditTaskCommandBehavior(helper.generateAddTaskCommand(toBeAdded),helper.generateListTaskCommand(),
-                helper.generateEditTaskCommand(toBeEdited),
+        assertEditTaskCommandBehavior(helper.generateAddFloatTaskCommand(toBeAdded), helper.generateAddFloatTaskCommand(toBeAdded2),helper.generateListTaskCommand(),
+                helper.generateEditTaskCommand(toBeEdited,1),
                 String.format(EditTaskCommand.MESSAGE_DUPLICATE_TASK, toBeEdited),
                 expectedAB,
                 expectedAB.getTaskList());
```

Comment:
This test still doesnt work the way it should. As I've mentioned in the chat group, the tasks are passed by reference when used as a parameter. Hence, both your expectedAB (taskbook) and your model is pointing to the same two tasks, toBeAdded and toBeAdded2. That is why when you execute edit -t 1 on your model, the toBeAdded in your expectedAB changes also. Hence, your assertEditTaskCommandBehavior "works".

What you should do is create 2 instances of toBeAdded and toBeAdded2. Then add one each to the expectedAB and model respectively, so that they contain the same data but different instances. After that, before calling for assertEditTaskCommandBehavior, you need to call for expectedAB.editTask(toBeEdited, targetTask). Hence you will call for the editTask command directly in your taskbook. During the assertEditTaskCommandBehavior, when the logic manager executes edit -t 1, it will go through the whole process and call for the editTask command in the model's taskbook. If both taskbooks match, then your edit command works the way it should.